4/typedefs.js

```diff
@@ -0,0 +1,214 @@
+      }
+    },
+    todo: async (object, args, context) => {
+      const session = context.driver.session();
+
+      try{
+        const cypherQuery = 'MATCH (t:Todo)-[:ASSIGNED_TO]->(a:Assignee) ' +
+                            'WHERE t.id = $id RETURN t, a';
+
+        const result = await session.run(cypherQuery, { id: args.id });
+        const todo = result.records.map(record => record.get('t').properties)[0];
+        //const assignee = result.records.map(record => record.get('a').properties)[0];
+        //todo.assignee = assignee;
+        return todo;
+      } finally {
+        session.close();
+      }
+    },
+    todosOrderedByState: async (object, args, context) => {
+      const session = context.driver.session();
+
+      try {
+        let cypherQuery = 'MATCH (t:Todo) RETURN t ORDER BY t.state ';
+        if(args.sortOrder == 'DESC') {
+          cypherQuery += 'DESC';
+        }
+
+        const result = await session.run(cypherQuery);
+        const todos = result.records.map(record => record.get('t').properties);
+        return todos;
+      } finally {
+        session.close();
+      }
+    },
+    skipAndLimitTodos: async (object, args, context) => {
+      const session = context.driver.session();
+      const skip = args.skip || 0;
+      const limit = args.limit || 0;
+
+      try {
+        const cypherQuery = 'MATCH (t:Todo) RETURN t ORDER BY t.id SKIP $skip LIMIT $limit';
+
+        const result = await session.run(cypherQuery, {
+          skip: skip,
+          limit: limit
+        });
+        const todos = result.records.map(record => record.get('t').properties);
+        return todos;
+      } finally {
+        session.close();
+      }
+    }
+  },
+  Mutation: {
+    createTodo: async (object, args, context, resolveInfo) => {
+      const todoId = args.id || uuid();
+      const todoState = args.state || 'UNDONE';
+
+      const session = context.driver.session();
+      try {
+        const cypherCreation = 'CREATE (t:Todo { id: $todoId, message: $message, state: $state })';
+
+        await session.run(cypherCreation, {
+          todoId: todoId,
+          message: args.message,
+          state: todoState
+        });
+        return true;
+      }
+      finally {
+        session.close();
+      }
+    },
+    createAssignee: async (object, args, context, resolveInfo) => {
+      const assigneeId = args.id || uuid();
+
+      const session = context.driver.session();
+      try {
+        const cypherCreation = 'CREATE (a:Assignee { id: $assigneeId, name: $name })';
+
+        await session.run(cypherCreation, {
+          assigneeId: assigneeId,
+          name: args.name
+        });
+        return true;
+      }
+      finally {
+        session.close();
+      }
+    },
+    createRelationship: async (object, args, context, resolveInfo) => {
+      const session = context.driver.session();
+
+      try {
+        const cypherRelation = 'MATCH (a:Assignee), (t:Todo) ' +
```

Comment:
Use backticks for multi-line strings
```js
const multilineString = `With
backtikcks you can easily write
on many lines and also do string
interpolation like ${this}`
```

Reply:
Also, did you check if `MATCH (a:Assignee), (t:Todo)` creates a cartesian product? You can do:
```cypher
MATCH (a:Assignee)
MATCH (t:Todo)
```
to avoid that. Check your statements on http://localhost:7474/browser/.